python/ray/rllib/agents/sac/sac_policy_graph.py

```diff
@@ -0,0 +1,233 @@
+        tf.keras.backend.set_session(self.session)
+
+        # self.global_step = tf.train.get_or_create_global_step()
+
+        self._init_placeholders(observation_space, action_space)
+        self._init_models(observation_space, action_space)
+        self._init_losses()
+
+        self.loss_inputs = (
+            ("observations", self._observations_ph),
+            ("next_observations", self._next_observations_ph),
+            ("actions", self._actions_ph),
+            ("rewards", self._rewards_ph),
+            ("terminals", self._terminals_ph),
+            ("importance_weights", self._importance_weights_ph),
+        )
+
+        TFPolicyGraph.__init__(
+            self,
+            observation_space,
+            action_space,
+            self.session,
+            obs_input=self._observations_ph,
+            action_sampler=self.policy.actions([self._observations_ph]),
+            loss=self.total_loss,
+            loss_inputs=self.loss_inputs,
+            # TODO(hartikainen): what is this for?
+            update_ops=None)
+
+        self.session.run(tf.global_variables_initializer())
+
+        self.update_target(tau=1.0)
+
+    def _init_placeholders(self, observation_space, action_space):
+        observation_shape = observation_space.shape
+        action_shape = action_space.shape
+
+        self._iteration_ph = tf.placeholder(tf.int64, (), name="iteration")
+
+        self._observations_ph = tf.placeholder(
+            tf.float32, (None, *observation_shape), name="observations")
+
+        self._next_observations_ph = tf.placeholder(
+            tf.float32, (None, *observation_shape), name="next_observations")
+
+        self._actions_ph = tf.placeholder(
+            tf.float32, (None, *action_shape), name="actions")
+
+        self._rewards_ph = tf.placeholder(
+            tf.float32, (None, 1), name="rewards")
+
+        self._terminals_ph = tf.placeholder(
+            tf.bool, (None, 1), name="terminals")
+
+        self._importance_weights_ph = tf.placeholder(
+            tf.float32, (None, 1), name="importance_weights")
+
+    def _init_models(self, observation_space, action_space):
+        """Initialize models for value-functions and policy."""
+        policy_type = self.config['policy']['type']
+        assert policy_type == 'GaussianLatentSpacePolicy', policy_type
+        policy_kwargs = self.config['policy']['kwargs']
+
+        self.policy = GaussianLatentSpacePolicy(
+            input_shape=observation_space.shape,
+            output_shape=action_space.shape,
+            **policy_kwargs)
+
+        self.log_alpha = tf.get_variable(
+            'log_alpha',
+            dtype=tf.float32,
+            initializer=0.0)
+        self.alpha = tf.exp(self.log_alpha)
+
+        Q_type = self.config['Q']['type']
+        # TODO(hartikainen): implement twin q
+        assert Q_type == 'FeedforwardQ', Q_type
+        Q_kwargs = self.config['Q']['kwargs']
+        self.Q = feedforward_model(
```

Comment:
I think it's necessary to call ModelCatalog.get_model() here, in order to support the custom models API.

To default to something besides the rllib built in models (i.e. the current feedforward keras models), you could do the same thing QMIX does: check if "custom_model"  is defined, and if so use the user-provided model, otherwise use a sac-specific default. That way, you can have a different default model, but the user can still plug in custom ones with the custom model API.